pkg/controller/persistentvolume/persistentvolume_controller.go

```diff
@@ -1152,6 +1158,13 @@ func (ctrl *PersistentVolumeController) isVolumeReleased(volume *api.PersistentV
 func (ctrl *PersistentVolumeController) scheduleOperation(operationName string, operation func(arg interface{}), arg interface{}) {
 	glog.V(4).Infof("scheduleOperation[%s]", operationName)
 
+	// Poke test code that an operation is just about to get started.
+	if ctrl.testingSchedulerOperationCond != nil {
```

Comment:
This is pretty ugly.  Can we do better?  What if we could add a function callback which we called here.  By default it would be nil (or an empty func) but test code could override that.